lab-traceyr/gulpfile.js

```diff
@@ -0,0 +1,116 @@
+const gulp = require('gulp');
+const eslint = require('gulp-eslint');
+const mocha = require('gulp-mocha');
+
+var testFiles = ['test/**/*.js'];
+var appFiles = ['lib/**/*.js'];
+var gulpFiles = ['gulpfile.js'];
+
+gulp.task('lint:app', () => {
+  gulp.src(appFiles)
+  .pipe(eslint({
+    rules: {
+      'no-console': 0,
+      'indent': [2, 2],
+      'quotes': [2, 'single'],
+      'linebreak-style': [2, 'unix'],      'semi': [2,'always']
+    },
+    envs: [
+      'node',
+      'es6',
+      'browser'
+    ],
+    globals: [
+      'describe',
+      'it',
+      'beforeEach',
+      'afterEach',
+      'before',
+      'after'
+    ],
+    ecmaFeatures: {
+      'modules': true,
+      'experimentalObjectRestSpread': true,
+      'impliedStrict': true
+    },
+    extends: 'eslint:recommended'
+  }))
+  .pipe(eslint.format());
+});
+
+gulp.task('lint:test', () => {
+  gulp.src(testFiles)
+  .pipe(eslint({
+    rules: {
+      'no-console': 0,
+      'indent': [2, 2],
+      'quotes': [2, 'single'],
+      'linebreak-style': [2, 'unix'],      'semi': [2,'always']
+    },
+    envs: [
+      'node',
+      'es6',
+      'browser'
+    ],
+    globals: [
+      'describe',
+      'it',
+      'beforeEach',
+      'afterEach',
+      'before',
+      'after'
+    ],
+    ecmaFeatures: {
+      'modules': true,
+      'experimentalObjectRestSpread': true,
+      'impliedStrict': true
+    },
+    extends: 'eslint:recommended'
+  }))
+  .pipe(eslint.format());
+});
+
+gulp.task('lint:gulp', () => {
+  gulp.src(gulpFiles)
+  .pipe(eslint({
+    rules: {
+      'no-console': 0,
+      'indent': [2, 2],
+      'quotes': [2, 'single'],
+      'linebreak-style': [2, 'unix'],      'semi': [2,'always']
+    },
+    envs: [
+      'node',
+      'es6',
+      'browser'
+    ],
+    globals: [
+      'describe',
+      'it',
+      'beforeEach',
+      'afterEach',
+      'before',
+      'after'
+    ],
+    ecmaFeatures: {
+      'modules': true,
+      'experimentalObjectRestSpread': true,
+      'impliedStrict': true
+    },
+    extends: 'eslint:recommended'
+  }))
+  .pipe(eslint.format());
+});
+
```

Comment:
If your linting tasks do exactly the same thing for each file, then you should just have one linting task and pass in all of the files to the task in the gulp.src array to keep your code more DRY (i.e. var files = ['test/**/*.js', 'lib/**/*.js', 'gulpfile.js'].